activerecord/lib/active_record/relation/calculations.rb

```diff
@@ -167,7 +183,7 @@ def pluck(*column_names)
       else
         relation = spawn
         relation.select_values = column_names.map { |cn|
-          @klass.has_attribute?(cn) || @klass.attribute_alias?(cn) ? arel_attribute(cn) : cn
+          @klass.effective_column_name?(cn) ? arel_attribute(cn) : cn
         }
         result = klass.connection.select_all(relation.arel, nil, bound_attributes)
         result.cast_values(klass.attribute_types)
```

Comment:
Duplicate pluck/unsafe_raw_pluck implementation suggests they may be a single method with a keyword arg indicating whether attrs are checked, defaulting to `ActiveRecord::Base.guard_unsafe_raw_sql`.